packages/cli/src/commands/seed.ts

```diff
@@ -0,0 +1,504 @@
+  type: z.literal('SimpleAllowList'),
+  owner: AddressSchema,
+  allowed: z.array(AddressSchema),
+});
+
+export const AllowListIncentiveSchema = z.object({
+  type: z.literal('AllowListIncentive'),
+  allowList: AddressSchema,
+  limit: z.coerce.bigint(),
+});
+
+export const ERC20IncentiveSchema = z.object({
+  type: z.literal('ERC20Incentive'),
+  asset: AddressSchema,
+  mintableMockAsset: z.boolean().default(true),
+  strategy: z.nativeEnum(StrategyType),
+  reward: z.coerce.bigint(),
+  limit: z.coerce.bigint(),
+  manager: AddressSchema.optional(),
+});
+
+export const ERC20VariableIncentiveSchema = z.object({
+  type: z.literal('ERC20VariableIncentive'),
+  asset: AddressSchema,
+  mintableMockAsset: z.boolean().default(true),
+  reward: z.coerce.bigint(),
+  limit: z.coerce.bigint(),
+  manager: AddressSchema,
+});
+
+export const IncentiveCriteriaSchema = z.object({
+  criteriaType: z.nativeEnum(SignatureType),
+  signature: zAbiItemSchema,
+  fieldIndex: z.number().nonnegative(),
+  targetContract: AddressSchema,
+});
+
+export const ERC20VariableCriteriaIncentiveSchema = z.object({
+  type: z.literal('ERC20VariableCriteriaIncentive'),
+  asset: AddressSchema,
+  mintableMockAsset: z.boolean().default(true),
+  reward: z.coerce.bigint(),
+  limit: z.coerce.bigint(),
+  manager: AddressSchema.optional(),
+  criteria: IncentiveCriteriaSchema,
+});
+
+export const BoostSeedConfigSchema = z.object({
+  protocolFee: z.coerce.bigint(),
+  maxParticipants: z.coerce.bigint(),
+  budget: z.union([AddressSchema, ManagedBudgetSchema]),
+  action: z.union([AddressSchema, EventActionSchema]),
+  validator: z.union([AddressSchema, SignerValidatorSchema]),
+  allowList: z.union([
+    AddressSchema,
+    SimpleDenyListSchema,
+    SimpleAllowListSchema,
+  ]),
+  incentives: z.array(
+    z.union([
+      AllowListIncentiveSchema,
+      ERC20IncentiveSchema,
+      ERC20VariableCriteriaIncentiveSchema,
+      ERC20VariableIncentiveSchema,
+    ]),
+  ),
+});
+
+async function getAllowList(
+  { allowList }: z.infer<typeof BoostSeedConfigSchema>,
+  { core }: { core: BoostCore },
+): Promise<AllowList> {
+  if (typeof allowList === 'string' && isAddress(allowList))
+    return await allowListFromAddress(
+      //@ts-expect-error i do what i want
+      { config: core._config, account: core._account },
+      allowList,
+    );
+  switch (allowList.type) {
+    case 'SimpleAllowList':
+      return core.SimpleAllowList(allowList as SimpleAllowListPayload);
+    case 'SimpleDenyList':
+      return core.SimpleDenyList(allowList as SimpleDenyListPayload);
+    default:
+      throw new Error('unusupported AllowList: ' + allowList);
+  }
+}
+
+async function fundBudget(
+  options: DeployableTestOptions,
+  erc20: MockERC20,
+  budget: Budget,
+  amount = parseEther('110'),
+) {
+  //await fundErc20(options, erc20, [], parseEther(amount.toString()))();
+  console.log(`minting ${amount} to ${options.account.address}`);
```

Comment:
```suggestion
```
this'll mess with the stdout and end up looking lik
```sh
minting 10 to 9x
{
 "boostIds": [0]
}
```

Reply:
ah got it